src/script/lua_api/l_item.cpp

```diff
@@ -424,10 +433,7 @@ int LuaItemStack::create(lua_State *L, const ItemStack &item)
 
 LuaItemStack* LuaItemStack::checkobject(lua_State *L, int narg)
 {
-	luaL_checktype(L, narg, LUA_TUSERDATA);
-	void *ud = luaL_checkudata(L, narg, className);
-	if(!ud) luaL_typerror(L, narg, className);
-	return *(LuaItemStack**)ud;  // unbox pointer
+	return *(LuaItemStack **)luaL_checkudata(L, narg, className);
```

Comment:
`luaL_checktype` is not needed because `luaL_checkudata` will check the type.
The `luaL_typerror` is dead code because `luaL_checkudata` will throw an error if there is no ItemStack.